script.js

```diff
@@ -1,4 +1,259 @@
+// There will be only two players. One human and one computer (for the Base solution).
+// The computer will always be the dealer.
+// Each player gets dealt two cards to start. (both face up, dealer has 1 card open and 1 card closed)
+// The player goes first, and decides if they want to hit (draw a card) or stand (end their turn).
+// The dealer has to hit if their hand is below 17.
+// Each players' score is the total of their card ranks. Jacks/Queen/Kings are 10. Aces can be 1 or 11.
+// The player who is closer to, but not above 21 wins the hand.
+
+//global variables
+var deck = [];
+var gameMode = "first round";
+var winningMode = ""; //blackjack, bigger number max 21
+var playerHand = []; //to reset
+var computerHand = [];
+var gameRound = 1;
+
+function getRandomNumber(max) {
+  return Math.floor(Math.random() * max);
+}
+function shuffleDeck() {
+  for (i = 0; i < deck.length; i += 1) {
+    var randomIndex = getRandomNumber(deck.length);
+    var currentIndex = i;
+    var currentCard = deck[currentIndex];
+    var randomCard = deck[randomIndex];
+    deck[currentIndex] = randomCard;
+    deck[randomIndex] = currentCard;
+  }
+  return deck;
+}
+
+function getStartingHand(numberOfCards, array) {
+  for (i = 0; i < numberOfCards; i += 1) {
+    var card = dealCard();
+    array.push(card);
+  }
+  return array;
+}
+
+function dealCard() {
+  var drawnCard = deck.pop();
+  return drawnCard;
+}
+
+function displayUserCards(array) {
+  var output = `drew: <br>`;
+  for (i = 0; i < array.length; i += 1) {
+    output = output + `${array[i].name} of ${array[i].suit}<br>`;
+  }
+  return output;
+}
+
+function displayAllPlayerCards() {
+  var output = `Player ${displayUserCards(
+    playerHand
+  )} <br>Computer ${displayUserCards(computerHand)}`;
+  return output;
+}
+
+// find total to determine ace rank and to determine whether dealer draws a card
+function findCardTotal(array) {
+  var cardTotal = 0;
+  for (i = 0; i < array.length; i += 1) {
+    cardTotal += array[i].rank;
+  }
+  return cardTotal;
+}
+
+function newAceTotal(array) {
+  var cardTotal = findCardTotal(array);
+  for (i = 0; i < array.length; i += 1) {
+    if (array[i].rank === 1) {
+      array[i].rank = 11;
+      var index = i;
+      cardTotal = findCardTotal(array);
+      if (cardTotal > 21) {
+        array[index].rank = 1;
+        cardTotal = findCardTotal(array);
+      }
+    }
+  }
+  return cardTotal;
+}
+
+function isMatch(number, cardTotal) {
+  return cardTotal === number;
+}
+
+function isLessThan(number, array) {
+  var cardTotal = newAceTotal(array);
+  return cardTotal < number;
+}
+
+function isBust(number, cardTotal) {
```

Comment:
I don't think you would want to take in the number. Ideally, the number should be fixed
`return cardTotal > 21`